tests/files/zrealm_std6.gno

```diff
@@ -0,0 +1,20 @@
+// PKGPATH: gno.land/r/std_test
+package std_test
+
+import (
+	"fmt"
+	"std"
+)
+
+func main() {
+	println(std.GetOrigCaller())
+	std.ExecAsPkg(func() {
+		println(std.GetOrigCaller())
+	})
+	println(std.GetOrigCaller())
+}
+
+// Output:
```

Comment:
We need more tests for this feature: multiple levels; reused func pointers; touching a variable outside of the scope; panic; probably more.